src/js/network/index.coffee

```diff
@@ -233,10 +233,31 @@ exports.connectivityToGraph = (cm, positions) ->
   for i in [0...cm.length]
     node = graph.addNode(positions[i])
     node.label = getLabel(node.index)
+    
   for i in [0...cm.length]
     for j in [0...cm[i].length]
-      if cm[i][j]
-        graph.addEdge(i, j)
+
+      # for removing nodes with only inputs or only outputs
+      row_minus_self = cm[i].reduce((a, b) -> a + b) - cm[i][i]
+      column = []
+      for i2 in [0...cm.length]
+        column.push(cm[i2][i])
+      col_minus_self = column.reduce((a, b) -> a + b) - cm[i][i]
+
+      if cm[i][j] # if there's a connection from i to j
+        if i==0 or i==1 or i==cm.length-1 or i==cm.length-2 #show all sensor/motor edges
```

Comment:
Hi Josh!

Equality should be checked with `is`, which compiles to `===`, unless you really need the `==` semantics (which should be considered harmful, IMHO). The `==` implicitly performs a type conversion; here it's obviously fine, but always using strict equality is a good habit, since in some cases loose equality can cause subtle bugs.